evm/utils/fixture_tests.py

```diff
@@ -287,37 +290,34 @@ def generate_fixture_tests(metafunc,
 #
 # Fixture Normalizers
 #
-def normalize_env(env):
-    return {
-        'currentCoinbase': decode_hex(env['currentCoinbase']),
-        'currentDifficulty': to_int(env['currentDifficulty']),
-        'currentNumber': to_int(env['currentNumber']),
-        'currentGasLimit': to_int(env['currentGasLimit']),
-        'currentTimestamp': to_int(env['currentTimestamp']),
-        'previousHash': decode_hex(env.get('previousHash', '00' * 32)),
-    }
-
-
 @to_dict
 def normalize_unsigned_transaction(transaction, indexes):
     yield 'data', decode_hex(transaction['data'][indexes['data']])
+    yield 'to', normalize_to_address(transaction['to'])
     yield 'gasLimit', to_int(transaction['gasLimit'][indexes['gas']])
     yield 'gasPrice', to_int(transaction['gasPrice'])
-    yield 'nonce', to_int(transaction['nonce'])
 
     if 'secretKey' in transaction:
         yield 'secretKey', decode_hex(transaction['secretKey'])
+        is_main_transaction = True
     elif 'v' in transaction and 'r' in transaction and 's' in transaction:
         yield 'vrs', (
             to_int(transaction['v']),
             to_int(transaction['r']),
             to_int(transaction['s']),
         )
+        is_main_transaction = True
     else:
-        raise KeyError("transaction missing secret key or vrs values")
+        is_main_transaction = False
```

Comment:
I think it will help prevent future errors if we validate the shard transaction fields. Something like:

```py

def is_valid_shard_transaction(transaction):
    return all(key in transaction for key in {'shardID', 'accessList'})

if 'secretKey' in transaction:
...
elif is_valid_shard_transaction(transaction):
     is_main_transaction = False
else:
    raise ValueError("transaction must be a signed main chain transaction, or a shard transaction.")